contracts/staking/contracts/src/vaults/EthVault.sol

```diff
@@ -0,0 +1,108 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+import "@0x/contracts-utils/contracts/src/LibSafeMath.sol";
+import "../interfaces/IEthVault.sol";
+import "./MixinVaultCore.sol";
+
+
+/// @dev This vault manages ETH.
+contract EthVault is
+    IEthVault,
+    MixinVaultCore
+{
+    using LibSafeMath for uint256;
+
+    // mapping from Owner to ETH balance
+    mapping (address => uint256) internal balances;
+
+    /// @dev Constructor.
+    constructor()
+        public
+    {} // solhint-disable-line no-empty-blocks
+
+    /// @dev Deposit an `amount` of ETH from `owner` into the vault.
+    /// Note that only the Staking contract can call this.
+    /// Note that this can only be called when *not* in Catostrophic Failure mode.
+    /// @param owner of ETH Tokens.
+    function depositFor(address owner)
+        external
+        payable
+    {
+        // update balance
+        uint256 amount = msg.value;
+        balances[owner] = balances[owner].safeAdd(msg.value);
```

Comment:
We might as well use `amount` here since `msg.value` was cached there, right? This said, `CALLVALUE` is typically actually cheaper than using a stack variable (it's a Gbase opcode), so I'd be in favor of just using `msg.value` for both uses and removing `amount` entirely, unless there is a readability concern.